packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -723,6 +723,18 @@ class IOSDevice extends Device {
       return LaunchResult.failed();
     } finally {
       startAppStatus.stop();
+
+      if ((isCoreDevice || forceXcodeDebugWorkflow) && debuggingOptions.debuggingEnabled && package is BuildableIOSApp) {
+        // When debugging via Xcode, after the app launches, reset the Generated
+        // settings to not include the custom configuration build directory.
+        // This is to prevent confusion if the project is later ran via Xcode
+        // rather than the Flutter CLI.
+        await updateGeneratedXcodeProperties(
+          project: FlutterProject.current(),
+          buildInfo: debuggingOptions.buildInfo,
+          targetOverride: mainPath,
+        );
+      }
```

Comment:
This is just moved so that it resets the settings after the Dart VM url has been found, rather than directly after starting the Xcode debug process. 

There wasn't any reason to indicate that this was the cause of the issue, but it doesn't hurt to do it a little later, so just to be safe I moved it.